utils/report_generator.py

```diff
@@ -0,0 +1,197 @@
+        DICOM_SENT (int): the number of DICOM images successfully sent to the server
+
+    Returns:
+        report_list (list): The list of items to append to the report .csv file
+    """
+    
+    fw_path = make_flywheel_path(fw, acq_id, file_name)
+    
+    if DICOM_PRESENT > DICOM_SENT:
+        status = 'Incomplete'
+    elif DICOM_PRESENT < DICOM_SENT:
+        status = 'Error'
+    else:
+        status = 'Complete'
+    
+    report_list = [acq_id, fw_path, file_name, DICOM_PRESENT, DICOM_SENT, status]
+    
+    return(report_list)
+    
+
+
+def generate_report(fw, acq_id, file_name, DICOMS_PRESENT,
+                    DICOMS_SENT, work_dir=Path('/flywheel/v0/report')):
+    """Writes status information of the current DICOM upload to a report .csv file
+    
+    Args:
+        fw (flywheel.Client()): the flywheel client for the current session
+        acq_id (str): the flywheel ID of the parent acquisition of the target file
+        file_name (str): the name of the target file
+        DICOM_PRESENT (int): the number of DICOM images present in the target file
+        DICOM_SENT (int): the number of DICOM images successfully sent to the server
+        work_dir (pathlib.PosixPath): The directory to generate/look for an existing report in
+
+    """
+    
+    if not os.path.exists(work_dir):
+        os.makedirs(work_dir)
+    
+    report_file = Path(work_dir/'dicom-send-report.csv')
+    
+    if not os.path.exists(report_file):
+        log.info('Initializing report')
+        initialize_report(report_file)
+    
+    log.debug('Writing to report')
+    new_row = generate_list(fw, acq_id, file_name, DICOMS_PRESENT, DICOMS_SENT)
+    log.info(new_row)
+    append_list_as_row(report_file, new_row)
+    
+
+def print_report(report_file):
+    """Print the current report .csv file in a human-readable way to the log
+    
+    Args:
+        report_file (pathlib.PosixPath): the .csv file to print
+
+
+    """
+    
+    # This gets the lengths of each element in each row/col.  These lengths are then
+    # used to format the printed output so that it's human readable.
+    with open(report_file, 'r', newline='') as read_obj:
+        csv_f = reader(read_obj)
+        lens = [ [len(i) for i in row] for row in csv_f]
+    
+    max_lens = [max(idx) for idx in zip(*lens)]
+    # 4 spaces are added between columns (1/2 a standard tab)
+    format_string = ["{:<"+str(ml+4)+"}" for ml in max_lens]
+    format_string = " ".join(format_string)+"\n"
+        
+    with open(report_file, 'r', newline='') as read_obj:
+        csv_f = reader(read_obj)
+        print_string = 'REPORT_SUMMARY:\n'
+        for row in csv_f:
+            print_string+=format_string.format(*row)
+        
+        log.info(print_string)
+
+
+def upload_report(fw, ses_id, acq_id=None,
+                  report_file=Path('/flywheel/v0/report/dicom-send-report.csv')):
+    """Upload a target report .csv file to a session as an attachment
```

Comment:
NIT: If the arguments of a function need to be split for line length, I find it is easier to read if each argument is on a newline.

Reply:
Again, a few places this occurs for functions in this script.